include/nil/blueprint/statistics.hpp

```diff
@@ -0,0 +1,121 @@
+                component_witness = component_w;
+                component_finished = component_f;
+            }
+
+            component_statistics () {
+            }
+
+            void component_call() {
+                component_counter++;
+            }
+        };
+
+        struct component_calls {
+
+            std::map<std::string, component_statistics> components;
+
+            std::set<std::string> unfinished_components = {
+                "non_native fp12 multiplication",
+                "is_in_g1",
+                "is_in_g2",
+                "native_bls12_381_pairing",
+                "hash to curve",
+                "comparison",
+            };
+
+            void add_record(std::string name, std::size_t rows, std::size_t gates, std::size_t witness) {
+
+                bool component_finished;
+
+                if (unfinished_components.find(name) != unfinished_components.end()) {
+                    component_finished = false;
+                } else {
+                    component_finished = true;
+                }
+
+                if (components.find(name) == components.end()) {
+                    components[name] = component_statistics(rows, gates, witness, component_finished);
+                } else {
+                    components[name].component_call();
+                }
+            }
+
+            void print() {
+                std::cout << "================\n";
+                std::cout << "statistics:\n";
+
+                std::size_t circuit_rows_amount = 0;
+
+                for (const auto& [name, component] : components) {
+                    circuit_rows_amount += (component.component_rows * component.component_counter);
+                }
+
+                std::cout << "total rows amount estimation: " << circuit_rows_amount << "\n";
+                std::cout << "________________\n";
+
+                for (const auto& [name, component] : components) {
+                    std::cout << "component name: " << name << "\n";
+                    std::cout << "Component was used " << component.component_counter << " times\n";
+                    if (!component.component_finished) {
+                        std::cout << "WARNING: component contains experimental features. Parameters are subject to change, do not fully trust to numbers below.\n";
+                    }
+                    std::cout << "gates amount: " << component.component_gates << "\n";
+                    std::cout << "witness size: " << component.component_witness << "\n";
+                    std::cout << "rows amount:  " << component.component_rows;
+                    std::size_t total_rows = component.component_rows * component.component_counter;
+                    std::cout << " (" << total_rows << " in total)\n";
+                    std::cout << "________________\n";
+                }
+                std::cout << std::endl;
+
+            }
+        };
+
+        nil::blueprint::component_calls statistics;
```

Comment:
global variable is not so good solution, can we keep `statistics` as member of `parser` and pass by reference like `assignment table`?